core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -59,10 +59,22 @@
      * @param negativeQuery the negative query for this boosting query.
      */
     public BoostingQueryBuilder(QueryBuilder positiveQuery, QueryBuilder negativeQuery) {
+        if (positiveQuery == null) {
+            throw new IllegalArgumentException("inner clause [positive] cannot be null.");
+        }
+        if (negativeQuery == null) {
+            throw new IllegalArgumentException("inner clause [negative] cannot be null.");
+        }
         this.positiveQuery = positiveQuery;
         this.negativeQuery = negativeQuery;
     }
 
+    private BoostingQueryBuilder() {
```

Comment:
you can just instantinate the prototype with `EmptyQueryBuilder.INSTANCE` instead?